include/zephyr/net/dhcpv4.h

```diff
@@ -44,6 +44,121 @@ enum net_dhcpv4_state {
 
 /** @endcond */
 
+/**
+ * @brief DHCPv4 message types
+ *
+ * These enumerations represent RFC2131 defined msy type codes, hence
+ * they should not be renumbered.
+ *
+ * Additions, removald and reorders in this definition must be reflected
+ * within corresponding changes to net_dhcpv4_msg_type_name.
+ */
+enum net_dhcpv4_msg_type {
+	NET_DHCPV4_MSG_TYPE_DISCOVER	= 1,
+	NET_DHCPV4_MSG_TYPE_OFFER	= 2,
+	NET_DHCPV4_MSG_TYPE_REQUEST	= 3,
+	NET_DHCPV4_MSG_TYPE_DECLINE	= 4,
+	NET_DHCPV4_MSG_TYPE_ACK		= 5,
+	NET_DHCPV4_MSG_TYPE_NAK		= 6,
+	NET_DHCPV4_MSG_TYPE_RELEASE	= 7,
+	NET_DHCPV4_MSG_TYPE_INFORM	= 8,
+};
+
+#ifdef CONFIG_NET_DHCPV4_OPTION_CALLBACKS
+
+struct net_dhcpv4_option_callback;
+
+/**
+ * @typedef net_dhcpv4_option_callback_handler_t
+ * @brief Define the application callback handler function signature
+ *
+ * @param cb Original struct net_dhcpv4_option_callback owning this handler
+ * @param length The length of data returned by the server. If this is
+ *		 greater than cb->max_length, only cb->max_length bytes
+ *		 will be available in cb->data
+ * @param msg_type Type of DHCP message that triggered the callback
+ * @param iface The interface on which the DHCP message was received
+ *
+ * Note: cb pointer can be used to retrieve private data through
+ * CONTAINER_OF() if original struct net_dhcpv4_option_callback is stored in
+ * another private structure.
+ */
+typedef void (*net_dhcpv4_option_callback_handler_t)(struct net_dhcpv4_option_callback *cb,
+						     size_t length,
+						     enum net_dhcpv4_msg_type *msg_type,
```

Comment:
I think it'd be better to provide the value, not the pointer. We don't want the app to mess with its content.